crates/compiler/can/src/module.rs

```diff
@@ -704,7 +704,11 @@ pub fn canonicalize_module_defs<'a>(
                 // def pattern has no default expressions, so skip
                 let loc_expr = &mut declarations.expressions[index];
 
-                fix_values_captured_in_closure_expr(&mut loc_expr.value, &mut VecSet::default());
+                fix_values_captured_in_closure_expr(
+                    &mut loc_expr.value,
+                    &mut VecSet::default(),
+                    &mut VecMap::default(),
```

Comment:
could maybe re-use the allocations for the vecset and vecmap by lifting them outside of the loop and clearing them on each iteration?